cli/src/Diagnostic.ts

```diff
@@ -0,0 +1,3 @@
+export type Diagnostic = Error & {
+  severity: "info" | "warning" | "error";
+};
```

Comment:
I think rather than create our own logging system we should go with something like https://github.com/winstonjs/winston

Reply:
Good point. I'm trying to abstract all of that away from the plugins so that plugins have a simple interface with the CLI. I think on the implementation side (i.e. in onDiagnostic) I would use something like winston. Might rethink it anyway.